### PR TITLE
[OV] Apply output insertion transformation first for model with If node 

### PR DESCRIPTION
**Pull request by kshpv**
### Changes

Apply output transformation first to correctly take the tensor names.

### Reason for changes

This allows to use of these names to collect data from If inputs

### Related tickets

122100

### Tests

Manually model from the ticket was tested.

### CODE REVIEW

File: nncf/openvino/quantization/quantize_ifmodel.py [hunk 1]
```diff
@@ -258,7 +261,9 @@ def get_if_cond_input_name(model: ov.Model, if_node: NNCFNode) -> str:
         """
         name_to_node_mapping = {op.get_friendly_name(): op for op in model.get_ops()}
         ov_node = name_to_node_mapping[if_node.node_name]
-        return ov_node.input_values()[0].any_name
+        if not ov_node.inputs()[0].get_tensor().get_names():
+            ov_node.inputs()[0].get_tensor().set_names(set(["if_0_input"]))
```
Comment (ilyachur) [inline nncf/openvino/quantization/quantize_ifmodel.py:265]: Why do we need to add some additional hardcoded names?

OpenVINO allows to work with model inputs/outputs without names.

Comment (kshpv) [reply nncf/openvino/quantization/quantize_ifmodel.py]: This PR is a hot fix of ticket 122100. Current implementation fully relies on the fact that tensors have names. These names are used to map outputs to corresponding tensors. 
The ticket with updating the current approach to not using tensor names will be created.

Comment (ilyachur) [reply nncf/openvino/quantization/quantize_ifmodel.py]: Ok.

**Comment by kshpv** [comment]
Updated fix.
Also noticed that implementation of `_add_outputs_before_if_node` was not common - maed it common reusing `create_output_insertion_commands_if_node()` from `OVBackend`.

**Review by alexsu52 — APPROVED** [review]

**Pull request merged**